### PR TITLE
Linkwarden - Harmonize Script, Add Monolith & Bugfixing

### LINKED ISSUE #194: Update Linkwarden | fatal error
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

Starting the update process of Linkwarden via `update` oder `bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/linkwarden.sh)"` from within a clean, newly installed lxc, the error message below appears.

### What settings are you currently utilizing?

Default Settings

### Which Linux distribution are you employing?

Ubuntu 22.04

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

```
    __    _       __                           __
   / /   (_)___  / /___      ______ __________/ /__  ____
  / /   / / __ \/ //_/ | /| / / __ `/ ___/ __  / _ \/ __ \
 / /___/ / / / / ,<  | |/ |/ / /_/ / /  / /_/ /  __/ / / /
/_____/_/_/ /_/_/|_| |__/|__/\__,_/_/   \__,_/\___/_/ /_/

 ✓ Stopped Linkwarden
 |hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint: 
hint:   git config pull.rebase false  # merge (the default strategy)
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint: 
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.

[ERROR] in line 67: exit code 0: while executing command git pull
```

### Please provide detailed steps to reproduce the issue.

Start the update process of Linkwarden via `update` oder `bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/linkwarden.sh)"`.

### LINKED ISSUE #295: New install of Linkwarden does not include `monolith`
### Please verify that you have read and understood the guidelines.

yes

### A clear and concise description of the issue.

A new install of the Linkwarden LXC does not include the `monolith` executable.  It looks like support for monolith was added in 2.6.0.

### What settings are you currently utilizing?

Default Settings

### Which Linux distribution are you employing?

Ubuntu 22.04

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

_No response_

### Please provide detailed steps to reproduce the issue.

1) Run bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/linkwarden.sh)"
2) Create account and log in to Linkwarden
3) Save a URL
4) Run `journalctl` to watch the logs.  See in the logs:

> Nov 16 20:25:10 linkwarden yarn[26576]: [1] /bin/sh: 1: monolith: not found
> Nov 16 20:25:10 linkwarden yarn[26576]: [1] Uncaught Monolith error...

### PR DESCRIPTION
**Pull request by MickLesk**
## Description

Add a Big Update for Linkwarden
-> Remove Git
-> Change to RELEASE
-> Add Monolith
-> Add Cargo / make
-> Optimize Script for Update
-> Add Storage-Check

Fully Tested, Install, Update, Update with .txt / Update Without .txt (Release) 

Fixes # (issue)
Fixes https://github.com/community-scripts/ProxmoxVE/issues/295 (Monolith)
Fixes https://github.com/community-scripts/ProxmoxVE/issues/194 (Fatal Error)

## Type of change
Please check the relevant option(s):

- [x] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation)

### CODE REVIEW

**Review by Mellowlynx — APPROVED** [review]

File: install/linkwarden-install.sh [hunk 1]
```diff
@@ -34,22 +36,28 @@ $STD apt-get install -y nodejs
...
 msg_ok "Installed Node.js/Yarn"
 
-msg_info "Clone Linkwarden Repository"
-$STD git clone https://github.com/linkwarden/linkwarden.git /opt/linkwarden
-cd /opt/linkwarden
-msg_ok "Cloned Linkwarden Repository"
+msg_info "Installing Monolith"
+$STD cargo install monolith
+echo 'export PATH=~/.cargo/bin:$PATH' >>~/.bashrc
+export PATH=~/.cargo/bin:$PATH
```
Comment (havardthom) [inline install/linkwarden-install.sh:42]:
```suggestion
source ~/.bashrc
```
or `. ~/.bashrc`

File: install/linkwarden-install.sh [hunk 2]
```diff
@@ -66,12 +74,24 @@ if [[ "${prompt,,}" =~ ^(y|yes)$ ]]; then
...
   msg_ok "Installed Adminer"
 fi
 
 msg_info "Installing Linkwarden (Patience)"
+cd /opt
 RELEASE=$(curl -s https://api.github.com/repos/linkwarden/linkwarden/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
-echo "${RELEASE}" >/opt/${APPLICATION}_version.txt
+wget -q "https://github.com/linkwarden/linkwarden/archive/refs/tags/${RELEASE}.zip"
+unzip -q ${RELEASE}.zip
+mv linkwarden-${RELEASE:1} /opt/linkwarden
```
Comment (havardthom) [inline install/linkwarden-install.sh:93]: In other scripts we keep `v` in version tag outside of `RELEASE` variable

**Review by havardthom — APPROVED** [review]

**Comment by havardthom** [comment]
@MickLesk You resolved all comments but didn't change anything?

**Comment by MickLesk** [comment]
> @MickLesk You resolved all comments but didn't change anything?

Next PR, because ive pushed a big Change with next PR. I wanted to save myself the stashing and rewriting. It does work. Fix is also already built

**Pull request merged**